browser/resources/settings/brave_leo_assistant_page/model_config_ui.ts

```diff
@@ -123,6 +132,54 @@ export class ModelConfigUI extends ModelConfigUIBase {
     this.modelRequestName = e.target.value
   }
 
+  onContextSizeChange_(e: Event) {
+    const inputElement = e.target as HTMLInputElement;
+    const inputValue = inputElement.value.trim();
+
+    if (inputValue === "") {
+      this.contextSize = null;
+      return;
+    }
+
+    const contextSize = parseInt(inputValue, 10);
+
+    if (isNaN(contextSize)) {
+      this.contextSize = null;
+      return;
+    }
+
+    /**
+     * The maximum value for `contextSize` is set to 2,147,483,647.
+     * If changing, see also components\ai_chat\core\browser\constants.h
```

Comment:
`LINT.IfChange`/`LINT.ThenChange`

Reply:
but this seems overly fragile to me and the UI code is definitely not the right place for these checks. The model should validate the data, not the code that accesses the model.

Reply:
These aren't **the checks** to make sure the values are accurate; but they are an attempt to not let the front-end experience deviate too much from what is enforced in the backend.